lib/rules/no-unused-services.js

```diff
@@ -0,0 +1,135 @@
+  },
+
+  ERROR_MESSAGE: getErrorMessage,
+
+  create(context) {
+    const unusedServicesMapping = {};
+
+    return {
+      // foo: service('bar')
+      Property(node) {
+        if (emberUtils.isInjectedServiceProp(node)) {
+          const name = node.key.name;
+          unusedServicesMapping[name] = node;
+        }
+      },
+      // @service('bar') foo;
+      ClassProperty(node) {
+        if (emberUtils.isInjectedServiceProp(node)) {
+          const name = node.key.name;
+          unusedServicesMapping[name] = node;
+        }
+      },
+      // this.foo...
+      MemberExpression(node) {
+        if (isEmpty(unusedServicesMapping) || !propertyGetterUtils.isSimpleThisExpression(node)) {
+          return;
+        }
+        const name = node.property.name;
+        delete unusedServicesMapping[name];
+      },
+      // 1. get(this, 'foo...');
+      // 2. getProperties(this, [..., 'foo..', ...]); or getProperties(this, ..., 'foo..', ...);
+      // 3. this.get('foo...');
+      // 4. this.getProperties([..., 'foo..', ...]); or this.getProperties(..., 'foo..', ...);
+      CallExpression(node) {
+        if (isEmpty(unusedServicesMapping)) {
+          return;
+        }
+
+        if (node.callee.name === 'get' && types.isThisExpression(node.arguments[0])) {
+          // 1. get(this, 'foo...');
+          const name = splitValue(node.arguments[1].value);
+          delete unusedServicesMapping[name];
+        } else if (
+          node.callee.name === 'getProperties' &&
+          types.isThisExpression(node.arguments[0])
+        ) {
+          // 2. getProperties(this, [..., 'foo..', ...]); or getProperties(this, ..., 'foo..', ...);
+          const argArray = types.isArrayExpression(node.arguments[1])
+            ? node.arguments[1].elements
+            : node.arguments.slice(1);
+          for (const elem of argArray) {
+            const name = splitValue(elem.value);
+            delete unusedServicesMapping[name];
+          }
+        } else if (propertyGetterUtils.isThisGetCall(node)) {
+          // 3. this.get('foo...');
+          const name = splitValue(node.arguments[0].value);
+          delete unusedServicesMapping[name];
+        } else if (
+          types.isThisExpression(node.callee.object) &&
+          node.callee.property.name === 'getProperties'
+        ) {
+          // 4. this.getProperties([..., 'foo..', ...]); or this.getProperties(..., 'foo..', ...);
+          const argArray = types.isArrayExpression(node.arguments[0])
+            ? node.arguments[0].elements
+            : node.arguments;
+          for (const elem of argArray) {
+            const name = splitValue(elem.value);
+            delete unusedServicesMapping[name];
+          }
+        }
+      },
+      // const { foo, ... } = this;
+      VariableDeclarator(node) {
+        if (isEmpty(unusedServicesMapping)) {
+          return;
+        }
+
+        if (types.isThisExpression(node.init) && types.isObjectPattern(node.id)) {
```

Comment:
This line will crash with the code: `let label;` (be sure to add a test case for this).